go.mod

```diff
@@ -1,35 +1,70 @@
 module github.com/keikoproj/addon-manager
 
-go 1.16
+go 1.17
 
 require (
-	cloud.google.com/go v0.65.0 // indirect
 	github.com/Masterminds/semver/v3 v3.1.1
-	github.com/go-logr/logr v0.4.0
-	github.com/go-logr/zapr v0.2.0 // indirect
-	github.com/google/go-cmp v0.5.2 // indirect
-	github.com/google/gofuzz v1.2.0 // indirect
-	github.com/google/uuid v1.1.2 // indirect
-	github.com/googleapis/gnostic v0.5.1 // indirect
-	github.com/imdario/mergo v0.3.11 // indirect
+	github.com/go-logr/logr v1.2.0
 	github.com/jinzhu/inflection v1.0.0
-	github.com/onsi/ginkgo v1.16.2
-	github.com/onsi/gomega v1.11.0
+	github.com/onsi/ginkgo v1.16.5
+	github.com/onsi/gomega v1.17.0
 	github.com/pkg/errors v0.9.1
-	github.com/prometheus/common v0.13.0 // indirect
-	github.com/spf13/cobra v1.0.0
-	go.uber.org/zap v1.15.0 // indirect
-	golang.org/x/crypto v0.0.0-20200820211705-5c72a883971a // indirect
-	golang.org/x/net v0.0.0-20201202161906-c7110b5ffcbb
-	golang.org/x/time v0.0.0-20200630173020-3af7569d3a1e // indirect
-	gomodules.xyz/jsonpatch/v2 v2.1.0 // indirect
+	github.com/spf13/cobra v1.2.1
+	golang.org/x/net v0.0.0-20210825183410-e898025ed96a
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
-	k8s.io/api v0.19.11
-	k8s.io/apiextensions-apiserver v0.19.11
-	k8s.io/apimachinery v0.19.11
-	k8s.io/client-go v0.19.11
-	k8s.io/klog/v2 v2.3.0 // indirect
-	k8s.io/kube-openapi v0.0.0-20200831175022-64514a1d5d59 // indirect
-	k8s.io/utils v0.0.0-20200821003339-5e75c0163111 // indirect
-	sigs.k8s.io/controller-runtime v0.6.5
+	k8s.io/api v0.23.0
```

Comment:
The ticket calls for v0.21, might want to double check this

Reply:
Yes, this should be `v0.21` which is one less than breaking changes.

Reply:
If we bump controller-runtime version, k8s.api version is bumped accordingly.

Reply:
Yes, we're saying change [all these dependencies](https://github.com/keikoproj/addon-manager/blob/c3ff3e626aa81547802fb051e009da6d6aed05db/go.mod#L27-L30) together to `v0.21`

Reply:
1d38f9fee5a09756fabee17ee3035ca31a25ef19